src/bin/nix-index.rs

```diff
@@ -327,6 +329,11 @@ async fn main() {
             .long("nixpkgs")
             .help("Path to nixpkgs for which to build the index, as accepted by nix-env -f")
             .default_value("<nixpkgs>"))
+        .arg(Arg::with_name("system")
+            .long("system")
```

Comment:
Maybe add a short option too?

```suggestion
            .short("s")
            .long("system")
```